### PR TITLE
Move test cases for delete recurring session

### PR DESCRIPTION
**Pull request by chwoozy**
* Moved test-cases from `UniqueStudentList` to `RecurringSession`
* Removed redundancies in checking for `RecurringSession`
* Added more test-cases for `delete_rec_session` command
* Removed potential non-isolated test cases in `DeleteRecurringSessionCommandTest`

### CODE REVIEW

**Review by JonahhGohh — APPROVED** [review]
LGTM

File: src/test/java/seedu/address/logic/commands/DeleteRecurringSessionCommandTest.java [hunk 1]
```diff
@@ -105,27 +104,47 @@ void execute_validNameValidIndexValidDate_success() {
...
+        Student exampleStudent1 = new StudentBuilder().withName("Father John").build();
+        Student copyExampleStudent1 = new StudentBuilder().withName("Father John").build();
+        Student exampleStudent2 = new StudentBuilder().withName("Mother Mary").build();
+        testingModel.addStudent(exampleStudent1);
+        comparedModel.addStudent(copyExampleStudent1);
+        testingModel.addStudent(exampleStudent2);
+        exampleStudent1.addSession(recurringSessionA);
+        copyExampleStudent1.addSession(copyRecurringSessionA);
+        exampleStudent2.addSession(recurringSessionA);
+        copyExampleStudent1.addSession(recurringSessionA);
```
Comment (samleewy) [inline src/test/java/seedu/address/logic/commands/DeleteRecurringSessionCommandTest.java]: maybe could group `testingModel`, `comparingModel` stuff together, a lil confusing to follow.

Comment (samleewy) [inline src/test/java/seedu/address/logic/commands/DeleteRecurringSessionCommandTest.java]: maybe could group `testingModel`, `comparingModel` stuff together, a lil confusing to follow.

**Review by samleewy — COMMENTED** [review]
some comments!

File: src/test/java/seedu/address/logic/commands/DeleteRecurringSessionCommandTest.java (hunk 1 above)
Comment (chwoozy) [reply src/test/java/seedu/address/logic/commands/DeleteRecurringSessionCommandTest.java]: Okay, will add more comments and reorder on the next commit!

Comment (chwoozy) [reply src/test/java/seedu/address/logic/commands/DeleteRecurringSessionCommandTest.java]: Okay, will add more comments and reorder on the next commit!

**Review by samleewy — APPROVED** [review]
father john and mother mary approves

**Pull request merged**